src/extension/importer/core/src/main/java/org/geoserver/importer/ImportContext.java

```diff
@@ -278,12 +279,48 @@ public void updated() {
         updateState();
     }
 
-    public void delete() throws IOException {
-        if (data != null) {
-            data.cleanup();
+    public boolean delete() throws IOException {
+        if (canBeRemoved(this)) {
+            for(ImportTask task : this.getTasks()) {
+                if(task.getData() != null) {
+                    if (canBePruned(task)) {
+                        task.getData().cleanup();
+                        return true;
+                    }
+                }
+            }
+        }
+        
+        return false;
+    }
+
+    protected boolean canBeRemoved(ImportContext context) {
+        if (context.getState() == ImportContext.State.COMPLETE || 
```

Comment:
The formatting is wrong in various places in this method. Also, ifs always need { and } around their blocks (even if one lineers)